deploy/olm-catalog/integreatly-operator/1.8.1/integreatly-operator.v1.8.1.clusterserviceversion.yaml

```diff
@@ -0,0 +1,362 @@
+          resources:
+          - groups
+          verbs:
+          - get
+          - create
+          - update
+          - delete
+        serviceAccountName: integreatly-operator
+      deployments:
+      - name: integreatly-operator
+        spec:
+          replicas: 1
+          selector:
+            matchLabels:
+              name: integreatly-operator
+          strategy: {}
+          template:
+            metadata:
+              labels:
+                name: integreatly-operator
+            spec:
+              containers:
+              - command:
+                - integreatly-operator
+                env:
+                - name: WATCH_NAMESPACE
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.annotations['olm.targetNamespaces']
+                - name: POD_NAME
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.name
+                - name: OPERATOR_NAME
+                  value: integreatly-operator
+                image: quay.io/integreatly/integreatly-operator:v1.8.1
+                imagePullPolicy: Always
+                name: integreatly-operator
+                resources: {}
+              serviceAccountName: integreatly-operator
+      permissions:
+      - rules:
+        - apiGroups:
+          - ""
+          resources:
+          - pods
+          - services
+          - endpoints
+          - persistentvolumeclaims
+          - events
+          - configmaps
+          - secrets
+          verbs:
+          - '*'
+        - apiGroups:
+          - ""
+          resources:
+          - namespaces
+          verbs:
+          - get
+        - apiGroups:
+          - apps
+          resources:
+          - deployments
+          - daemonsets
+          - replicasets
+          - statefulsets
+          verbs:
+          - '*'
+        - apiGroups:
+          - monitoring.coreos.com
+          resources:
+          - servicemonitors
+          verbs:
+          - get
+          - create
+        - apiGroups:
+          - apps
+          resourceNames:
+          - integreatly-operator
+          resources:
+          - deployments/finalizers
+          verbs:
+          - update
+        - apiGroups:
+          - integreatly.org
+          resources:
+          - '*'
+          verbs:
+          - '*'
+        serviceAccountName: integreatly-operator
+    strategy: deployment
+  installModes:
+  - supported: true
+    type: OwnNamespace
+  - supported: true
+    type: SingleNamespace
+  - supported: true
+    type: MultiNamespace
+  - supported: true
+    type: AllNamespaces
+  keywords:
+  - integreatly
+  labels:
+    alm-owner-integreatly: integreatly-operator
+    operated-by: integreatly-operator
+  maintainers:
+  - email: integreatly-support@redhat.com
+    name: Integr8ly
+  maturity: alpha
+  minKubeVersion: 1.11.0
+  provider:
+    name: Integr8ly
+  replaces: integreatly-operator.v1.8.0
```

Comment:
@philbrookes @maleck13 Do either of you have any more information about how this should work?

The tooling @dimitraz is using to generate these files looks to be assuming that you have all versions of the manifest, at least the previous version, in this repo so that it can use that to generate the new one. It also sets this replaces value. If it is expecting the previous manifest here, should we just have all versions here?

What's the best practise here for maintaining these? It's hard to find any examples that do it in the same way, but on the productised versions that i could find, all manifest versions were just kept in the operator repo.

Reply:
@mikenairn Yeah, I found one productised example that seemed to keep all the versions within the operator repo. If we keep on using the `gen-csv` command this shouldn't be very difficult to maintain either. 

If we don't need the replaces field, we can use a generic integreatly-operator csv template that the command can use to base off of and remove the replaces field on the generated csv, but that isn't necessarily the intended usage for the tool. I think it depends on what the requirements are for delorean